### DISCUSSION TITLE
429 Error:  Too Many Requests

### DISCUSSION THREAD

**Discussion by spowell-216** [iTwin/itwinjs-core#1491] [Q&A] [answered]
I have recently started getting this error sporadically over the last week.  I double-checked how i am connecting to the iHub Manager and i am following the API Docs for [Managing ClientRequestContext](https://www.itwinjs.org/learning/backend/managingclientrequestcontext/?term=clientrequestconte#managing-the-clientrequestcontext). 

I have tried on both 2.15.1 as well as 2.15.5

Info    |itwin-client.Request| https://buddi.bentley.com/WebService/GetUrl/?url=IMSOpenID&region=0
Trace   |itwin-client.Request| GET 200 https://buddi.bentley.com/WebService/GetUrl/?url=IMSOpenID&region=0 (305ms)
Trace   |imodel-agent| Request Context Received
Info    |itwin-client.Request| https://api.bentley.com/contextregistry/v2.5/Repositories/BentleyCONNECT--Main/ConnectedContext/Project?$select=*&$filter=Name+eq+'a valid project name'
Trace   |itwin-client.Request| GET 429 https://api.bentley.com/contextregistry/v2.5/Repositories/BentleyCONNECT--Main/ConnectedContext/Project?$select=*&$filter=Name+eq+'a valid project name' (343ms)
Error   |itwin-client.Request| 429 Error: Too Many Requests
Error   |Program| Not Enabled: Failed initializing iModelAgent:
Error: Failed getting Project ID from hub. Error: Too Many Requests

i have my Agent App registered per the [instructions](https://www.itwinjs.org/learning/tutorials/registering-applications/?term=register) with iModelAdmin permissions.

Any help would be much appreciated.

**Comment by calebmshafer** [comment, marked as answer]
@spowell-216 this is due to an update introduced in 2.15.0 to switch the endpoints used in all client libraries to the new one at `http://api.bentley.com`. The new endpoints have a fairly low rate limit applied for all older OIDC Clients and require an upgrade to the client in order to increase the limit.

We are working on resolving this issue, but for the moment would you be able to manually update the client?  This can be done by going to https://developer.bentley.com/my-apps/ and clicking on the specific client.  Once on the details page for the client, there should be an "Upgrade" button at the top of the page, if you click it and retry running your Agent the rate limit issues should go away.

Apologizes for the inconvenience and we will be updating the documentation to reflect this change. Thanks!

**Reply by spowell-216** [reply]
@calebmshafer that seems to have resolved the problem.  thanks for the quick response.
